src/main/java/com/ss/mqtt/broker/service/impl/FileCredentialsSource.java

```diff
@@ -0,0 +1,33 @@
+package com.ss.mqtt.broker.service.impl;
+
+import com.ss.mqtt.broker.exception.CredentialsSourceException;
+import org.jetbrains.annotations.NotNull;
+
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.util.Properties;
+
+public class FileCredentialsSource extends AbstractCredentialSource {
+
+    private final String fileName;
+
+    public FileCredentialsSource(@NotNull String fileName) {
```

Comment:
you marked the argument as not null but you use not required property for this

Reply:
I used getProperty with defaultValue. So everything is OK.